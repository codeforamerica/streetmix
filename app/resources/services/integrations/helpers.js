const { User } = require('../../../db/models')

/**
 finds the database record for the given user
 */
// might be better to do something besides return null
// we can actually probably get rid of this function if we want, but it might help
// keep some of the other code clean
const findUser = async function (userId) {
  const user = await User.findOne({ where: { auth0Id: userId } })
  if (user === null) {
    return null
  }
  return user.dataValues
}

// if the provider we're trying to find exists
// return its index value or -1 if not found
const addOrUpdateByProviderName = function (array, item) {
  const indexValue = array.findIndex(
    (_item) => _item.provider === item.provider
  )
  if (indexValue === -1) {
    array.push(item)
  } else {
    array[indexValue] = item
  }
}

/**
 given a user, check their identity data,
 depending on what data you find, add subscriber role

 TODO: make this less permissive(maybe add a status field or have logic depending on the provider), tested,
 and remove the role if we're confident we should (probably shouldn't worry about this for MVP)
 */
const syncAccountStatus = async function (userId) {
  const databaseUser = await User.findOne({ where: { auth0Id: userId } })
  if (databaseUser.identities.some((identity) => 'user_id' in identity)) {
    databaseUser.addRole('SUBSCRIBER_1')
  }
}

module.exports = { findUser, addOrUpdateByProviderName, syncAccountStatus }
