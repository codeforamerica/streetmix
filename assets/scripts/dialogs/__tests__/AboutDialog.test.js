/* eslint-env jest */
import React from 'react'
import { cleanup } from '@testing-library/react'
import { renderWithReduxAndIntl } from '../../../../test/helpers/render'
import AboutDialog from '../AboutDialog'

jest.mock('../About/credits.json', () => require('../About/__mocks__/credits.json'))

describe('AboutDialog', () => {
  afterEach(cleanup)

  it('renders snapshot', () => {
    const wrapper = renderWithReduxAndIntl(<AboutDialog />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
