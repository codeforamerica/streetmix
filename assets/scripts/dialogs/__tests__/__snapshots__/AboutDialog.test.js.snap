// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutDialog renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="dialog-box-container"
  >
    <div
      class="dialog-box-backdrop"
    />
    <div
      class="dialog-box"
      role="dialog"
    >
      <button
        class="close"
        title="Dismiss"
      >
        <svg
          class="svg-inline--fa fa-times"
        />
      </button>
      <div
        class="about-dialog"
      >
        <header>
          <div
            class="streetmix-logo"
          />
          <h1>
            <span>
              About Streetmix.
            </span>
          </h1>
        </header>
        <div
          class="dialog-content"
        >
          <div
            class="about-dialog-content"
          >
            <div
              class="about-dialog-left"
            >
              <p>
                <span>
                  Design, remix, and share your street. Add bike paths, widen sidewalks or traffic lanes, learn how all of this can impact your community.
                </span>
              </p>
              <p>
                <span>
                  Streetmix is generously sponsored by:
                </span>
              </p>
              <ul
                class="about-dialog-sponsors"
              >
                <li>
                  <a
                    href="https://codeforamerica.org/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <img
                      alt="Code for America"
                      src="/images/sponsors/codeforamerica.png"
                    />
                  </a>
                </li>
                <li>
                  <a
                    href="https://lyft.com/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <img
                      alt="Lyft"
                      src="/images/sponsors/lyft.svg"
                    />
                  </a>
                </li>
                <li>
                  <a
                    href="https://www.mozilla.org/en-US/moss/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <img
                      alt="Mozilla Open Source Support"
                      src="/images/sponsors/mozilla.svg"
                    />
                  </a>
                </li>
              </ul>
              <p>
                <a
                  href="https://opencollective.com/streetmix/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    Support us financially
                  </span>
                </a>
                <br />
                <a
                  href="https://github.com/streetmix/streetmix/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    We’re open source!‎
                  </span>
                </a>
                <br />
                <a
                  href="https://medium.com/streetmixology"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    Visit Streetmix blog
                  </span>
                </a>
                <br />
                <a
                  href="https://streetmix.readthedocs.io/en/latest/guidebook/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    Guidebook
                  </span>
                </a>
              </p>
              <p>
                <a
                  href="https://streetmix.net/terms-of-service/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    Terms of service
                  </span>
                </a>
                <br />
                <a
                  href="https://streetmix.net/privacy-policy/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    Privacy policy
                  </span>
                </a>
              </p>
            </div>
            <div
              class="about-dialog-right"
            >
              <h2>
                <span>
                  Project team
                </span>
              </h2>
              <div
                class="credits-team"
              >
                <div
                  class="team-member"
                >
                  <div
                    class="team-member-mugshot"
                    style="background-image: url(/images/team/dwight.jpg);"
                  />
                  <span
                    class="team-member-name"
                  >
                    Dwight Schrute
                  </span>
                  <span
                    class="team-member-title"
                  >
                    assistant to the regional manager
                  </span>
                </div>
                <div
                  class="team-member"
                >
                  <div
                    class="team-member-mugshot"
                    style="background-image: url(/images/team/michael.jpg);"
                  />
                  <span
                    class="team-member-name"
                  >
                    <a
                      href="https://dunder-mifflin.com"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Michael Scott
                    </a>
                  </span>
                  <span
                    class="team-member-title"
                  >
                    regional manager
                  </span>
                </div>
              </div>
              <h2>
                <span>
                  Past team members
                </span>
              </h2>
              <div
                class="credits-team credits-team-past"
              >
                <div
                  class="team-member"
                >
                  <div
                    class="team-member-mugshot"
                    style="background-image: url(/images/team/jim.jpg);"
                  />
                  <span
                    class="team-member-name"
                  >
                    <a
                      href="http://sabre.com"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Jim Halpert
                    </a>
                  </span>
                  <span
                    class="team-member-title"
                  >
                    sales
                  </span>
                </div>
                <div
                  class="team-member"
                >
                  <div
                    class="team-member-mugshot"
                    style="background-image: url(/images/team/pam.jpg);"
                  />
                  <span
                    class="team-member-name"
                  >
                    Pam Beesly
                  </span>
                  <span
                    class="team-member-title"
                  >
                    office administrator
                  </span>
                </div>
              </div>
              <div
                class="credits-container"
              >
                <div
                  class="credits-credits-left"
                >
                  <h3>
                    <span>
                      Advisors
                    </span>
                  </h3>
                  <ul>
                    <li>
                      Jan Levinson
                    </li>
                    <li>
                      Robert California
                    </li>
                  </ul>
                  <h3>
                    <span>
                      Additional illustrations
                    </span>
                  </h3>
                  <ul>
                    <li>
                      Kevin Malone
                    </li>
                    <li>
                      Meredith Palmer
                    </li>
                    <li>
                      Stanley Hudson
                    </li>
                  </ul>
                  <h3>
                    <span>
                      Additional code
                    </span>
                  </h3>
                  <ul>
                    <li>
                      Angela Martin
                    </li>
                    <li>
                      Oscar Martinez
                    </li>
                    <li>
                      Phyllis Lapin
                    </li>
                  </ul>
                  <h3>
                    <span>
                      Additional contributors
                    </span>
                  </h3>
                  <ul>
                    <li>
                      Creed Bratton
                    </li>
                    <li>
                      Kelly Kapoor, 
                      <i>
                        customer service
                      </i>
                    </li>
                    <li>
                      Toby Flenderson, 
                      <i>
                        human resources
                      </i>
                    </li>
                  </ul>
                  <h3>
                    <span>
                      Special thanks to
                    </span>
                  </h3>
                  <ul>
                    <li>
                      Andy Bernard, 
                      <i>
                        Cornell
                      </i>
                    </li>
                    <li>
                      Darryl Philbin
                    </li>
                    <li>
                      Kelly Erin Hannon
                    </li>
                    <li>
                      Ryan Howard, 
                      <i>
                        WUPHF
                      </i>
                    </li>
                    <li>
                      <h4>
                        Dunder Mifflin
                      </h4>
                      <ul>
                        <li>
                          David Wallace
                        </li>
                        <li>
                          Nellie Bertram
                        </li>
                        <li>
                          Holly Flax
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
                <div
                  class="credits-credits-right"
                >
                  <h3>
                    <span>
                      Translators
                    </span>
                  </h3>
                  <ul>
                    <li>
                      <h4>
                        <span>
                          Arabic
                        </span>
                      </h4>
                      <ul>
                        <li>
                          Charles Miner
                        </li>
                        <li>
                          Karen Filippelli
                        </li>
                      </ul>
                    </li>
                    <li>
                      <h4>
                        <span>
                          German
                        </span>
                      </h4>
                      <ul>
                        <li>
                          Gabe Lewis, 
                          <i>
                            Sabre
                          </i>
                        </li>
                        <li>
                          Jo Bennett, 
                          <i>
                            Sabre
                          </i>
                        </li>
                        <li>
                          Todd Packer
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="dialog-primary-action"
        >
          <span>
            Close
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
